libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1451,6 +1451,8 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 	auto const* intType = dynamic_cast<IntegerType const*>(&_type);
 	auto const* fixedType = dynamic_cast<FixedPointType const*>(&_type);
 
+	auto const fixedFactor = fixedType ? u256(pow(u256(10), fixedType->fractionalDigits())) : u256(0);
```

Comment:
From https://github.com/ethereum/solidity/pull/3389/files/089e000b86e7807bdf6dc358c303b4fe448c83aa..cfbc94dce4d087d103e62a64e62cca7ff5dfcefa#diff-07335c711d9742b9c53a75648c7ad55cR501 it could actually be that `10**fractionalDIgits` is `2**256` (not possible in this specific case, but still) - so perhaps rather use bigint for this?

Also, is the result of `pow` not a `u256` or why are you converting again?

Reply:
Oh and should it not be 1 for an integer instead of 0?

Reply:
I had just set it to zero for integers since it's not used for integers.

Reply:
And for the overflow, that is guaranteed not to happen due to the check in FixedPointType's constructor, but I'll use bigint anyway.

Reply:
Hm, does this zero lead to an error further down?

The reason I'm asking is that we should never ever assume any consistency between parts of the code that are more than 50 lines away from each other. Always use asserts to check any assumptions you might have about different parts of the code. This is very useful if different people work on the same code and it also helps with reviews since it catches errors that might be hidden because a reviewer did not think about parts of the code not visible in the changeset.